src/@context/Automation/AutomationProvider.tsx

```diff
@@ -0,0 +1,239 @@
+import Loader from '../../components/@shared/atoms/Loader'
+import { useMarketMetadata } from '../MarketMetadata'
+import DeleteAutomationModal from './DeleteAutomationModal'
+
+export interface NativeTokenBalance {
+  symbol: string
+  balance: string
+}
+export interface AutomationProviderValue {
+  autoWallet: Wallet
+  autoWalletAddress: string
+  isAutomationEnabled: boolean
+  balance: UserBalance
+  nativeBalance: NativeTokenBalance
+  isLoading: boolean
+  decryptPercentage: number
+  updateBalance: () => Promise<void>
+  setIsAutomationEnabled: (isEnabled: boolean) => void
+  deleteCurrentAutomationWallet: () => void
+  importAutomationWallet: (encryptedJson: string) => Promise<boolean>
+  hasValidEncryptedWallet: () => boolean
+  decryptAutomationWallet: (password: string) => Promise<boolean>
+}
+
+// Refresh interval for balance retrieve - 20 sec
+const refreshInterval = 20000
+
+// Context
+const AutomationContext = createContext({} as AutomationProviderValue)
+
+// Provider
+function AutomationProvider({ children }) {
+  const { address } = useAccount()
+  const { approvedBaseTokens } = useMarketMetadata()
+  const chainId = useChainId()
+  const { automationWalletJSON, setAutomationWalletJSON } = useUserPreferences()
+
+  const [autoWallet, setAutoWallet] = useState<Wallet>()
+  const [isAutomationEnabled, setIsAutomationEnabled] = useState<boolean>(false)
+  const [isLoading, setIsLoading] = useState<boolean>(false)
+  const [autoWalletAddress, setAutoWalletAddress] = useState<string>()
+  const [decryptPercentage, setDecryptPercentage] = useState<number>()
+
+  const { data: balanceNativeToken } = useBalance({
+    address: autoWallet?.address as `0x${string}`
+  })
+
+  const [nativeBalance, setNativeBalance] = useState<NativeTokenBalance>()
+  const [balance, setBalance] = useState<UserBalance>({})
+
+  const [hasDeleteRequest, setHasDeleteRequest] = useState(false)
+
+  const wagmiProvider = useProvider()
+
+  useEffect(() => {
+    if (!automationWalletJSON) setAutoWalletAddress(undefined)
+    else
+      setAutoWalletAddress(
+        ethers.utils.getJsonWalletAddress(automationWalletJSON)
+      )
+  }, [automationWalletJSON])
+
+  useEffect(() => {
+    if (autoWallet && !isAutomationEnabled) {
+      toast.info(`Automation disabled`)
+      return
+    }
+
+    if (autoWallet?.address)
+      toast.success(
+        `Successfully enabled automation wallet with address ${accountTruncate(
+          autoWallet?.address
+        )}`
+      )
+  }, [isAutomationEnabled, autoWallet])
+
+  const updateBalance = useCallback(async () => {
+    if (!autoWallet) return
+
+    try {
+      if (balanceNativeToken)
+        setNativeBalance({
+          symbol: balanceNativeToken?.symbol.toLowerCase() || 'ETH',
+          balance: balanceNativeToken?.formatted
+        })
+
+      if (approvedBaseTokens?.length > 0) {
+        const newBalance: UserBalance = {}
+        await Promise.all(
```

Comment:
suggestion: While using  `await Promise.all(...)`  efficient, if one of the promises fails, it will reject the entire batch. If we want to handle errors for each token individually and continue fetching balances for the others, we should consider using something like `Promise.allSettled()`.